tls-attestation/draft-fossati-tls-attestation.md

```diff
@@ -486,7 +493,98 @@ Auth | {CertificateVerify*}
+
+The attestation evidence is produced through the TPM2_Quote operation. The evidence along with all other relevant metadata is transmitted in a format derived from the {{WebAuthn}} Attestation Statements. This format and the workflows around it are defined below.
+
+### TPM Platform Attestation Statement Format
+
+The TPM Platform Attestation Statement is a modified version of the TPM Attestation Statement Format which covers key attestation tokens.
+
+~~~~
+    tpmPlatStmtFormat = {
+                          ver: "2.0",
+                          (
+                              alg: COSEAlgorithmIdentifier,
+                              x5c: [ pakCert: bytes, * (caCert: bytes) ]
+                          )
+                          sig: bytes,
+                          attestInfo: bytes,
+                        }
+~~~~
+{: #figure-tpm-plat-attest-stmt title="TPM Platform Attestation Statement Format"}
+
+- _ver_: The version of the TPM specification to which the signature conforms.
+- _alg_: A COSEAlgorithmIdentifier containing the identifier of the algorithm used to generate the attestation signature.
+- _x5c_: A certificate for the PAK, followed by its certificate chain (in X.509 encoding).
+  - _pakCert_: The PAK certificate used for the attestation, in X.509 encoding.
+- _sig_: The attestation signature, in the form of a TPMT_SIGNATURE structure as specified in Part 2, section 11.3.4 of {{TPM2.0}}.
+- _attestInfo_: The TPMS_ATTEST structure over which the above signature was computed, as specified in Part 2, section 10.12.8 of {{TPM2.0}}.
+
+### Signing Procedure
+
+Generate a signature using the operation specified in Part 3, section 18.4 of {{TPM2.0}}, using the PAK as the signing key, the out-of-band agreed-upon PCR selection. Freshness of the attestation is given by the nonce provided by the relying party. The nonce is included as qualified data to the TPM2_Quote operation, concatenated with an identifier of the platform being attested, as shown below:
+
+  _extraData_ = _platformUuid_ || _relyingPartyNonce_
+
+The platform identifier is a 16-bytes long UUID, with the remaining data representing the nonce. The UUID is intended to help the verifier link the platform with its expected reference values.
+
+Set the _attestInfo_ field to the quoted PCR selection produced by the operation, and _sig_ to the signature generated above.
+
+### Verification Procedure
+
+The inputs to the verification procedure are as follows:
+
+- an attestation statement in the format described above
+- the nonce sent to the attester
+- a database of reference values for various platforms
+
+The steps for verifying the attestation:
+
+- Verify that the attestation token is valid CBOR conforming to the CTAP2 form and perform CBOR decoding on it to extract the contained fields.
```

Comment:
I believe a reference to CTAP2 would be needed. Do you need everything from CTAP2 or only selected parts?

Reply:
I only need [this part](https://fidoalliance.org/specs/fido-v2.0-ps-20190130/fido-client-to-authenticator-protocol-v2.0-ps-20190130.html#message-encoding), relating to message encoding.